presto-bytecode/src/main/java/com/facebook/presto/bytecode/MethodDefinition.java

```diff
@@ -70,6 +74,15 @@ public MethodDefinition(
             Iterable<Parameter> parameters
     )
     {
+        // Code generation could failed if parameter count is 255 or more. Given that fixing this directly might be difficult, handle it by capping the maximum number of arguments
+        // to 250.
+        int parameterCount = Iterables.size(parameters);
+        checkArgument(
+                parameterCount <= MAX_ALLOWED_VARARG_ARGUMENTS,
+                "Number of arguments (%d) for vararg functions exceeds limit (%d)",
```

Comment:
you moved the check to the right place, but the message is till wrong. From the point of this class, there are no varargs. Anyway, you're _not_ producing end-user targeting error message (otherwise you'd use `PrestoException`), so why don't you phrase this simply as `"Number of arguments (%s) exceeds limit (%s)"` ?